src/tools/genpolicy/src/registry_containerd.rs

```diff
@@ -359,74 +347,41 @@ async fn get_verity_hash(
     Ok(verity_hash)
 }
 
-async fn create_verity_hash_file(
-    use_cached_files: bool,
+async fn create_decompressed_layer_file(
+    client: &containerd_client::Client,
     layer_digest: &str,
-    base_dir: &Path,
     decompressed_path: &Path,
     compressed_path: &Path,
-    verity_path: &Path,
-    client: &containerd_client::Client,
 ) -> Result<()> {
-    if use_cached_files && decompressed_path.exists() {
-        info!("Using cached file {:?}", &decompressed_path);
-    } else {
-        std::fs::create_dir_all(base_dir)?;
+    info!("Pulling layer {layer_digest}");
+    let mut file = tokio::fs::File::create(&compressed_path)
+        .await
+        .map_err(|e| anyhow!(e))?;
 
-        create_decompressed_layer_file(
-            use_cached_files,
-            layer_digest,
-            decompressed_path,
-            compressed_path,
-            client,
-        )
-        .await?;
-    }
+    info!("Decompressing layer");
 
-    do_create_verity_hash_file(decompressed_path, verity_path)
-}
+    let req = containerd_client::services::v1::ReadContentRequest {
+        digest: layer_digest.to_string(),
+        offset: 0,
+        size: 0,
+    };
+    let req = with_namespace!(req, "k8s.io");
+    let mut c = client.content();
+    let resp = c.read(req).await?;
```

Comment:
This is the reason we can't reuse create_decompressed_layer_file and other adjacent methods from registry.rs. Here we fetch the content in a different way (using containerd_client)